### PR TITLE
Fix regular expression for (.0*) in NumberValue, refs #753

### PR DESCRIPTION
**Pull request by mwjames**
refs #753, https://phabricator.wikimedia.org/T40476

### CODE REVIEW

**Comment by mwjames** [comment]
http://sandbox.semantic-mediawiki.org/wiki/Issue/1334

**Comment by mwjames** [comment]
@kghbln FYI

File: includes/datavalues/SMW_DV_Number.php [hunk 1]
```diff
@@ -71,10 +71,20 @@ static protected function parseNumberValue( $value, &$number, &$unit ) {
...
 		$kiloseparator = NumberFormatter::getInstance()->getThousandsSeparatorForContentLanguage();
 
-		$parts = preg_split( '/([-+]?\s*\d+(?:\\' . $kiloseparator . '\d\d\d)*' .
-		                      '(?:\\' . $decseparator . '\d+)?\s*(?:[eE][-+]?\d+)?)/u',
-		                      trim( str_replace( array( '&nbsp;', '&#160;', '&thinsp;', ' ' ), '', $value ) ),
-		                      2, PREG_SPLIT_DELIM_CAPTURE );
+		// #753
+		$regex = '/([-+]?\s*(?:' .
+				// Either numbers like 10,000.99 that start with a digit
+				'\d+(?:\\' . $kiloseparator . '\d\d\d)*(?:\\' . $decseparator . '\d+)?' .
```
Comment (thiemowmde) [inline includes/datavalues/SMW_DV_Number.php]: If you want to support numbers like "4." and "5.e-5" you can replace `(?:\\' . $decseparator . '\d+)?` with `(?:\\' . $decseparator . '\d*)?`.

File: includes/datavalues/SMW_DV_Number.php [hunk 2]
```diff
@@ -71,10 +71,20 @@ static protected function parseNumberValue( $value, &$number, &$unit ) {
 		$decseparator = NumberFormatter::getInstance()->getDecimalSeparatorForContentLanguage();
 		$kiloseparator = NumberFormatter::getInstance()->getThousandsSeparatorForContentLanguage();
```
Comment (thiemowmde) [inline includes/datavalues/SMW_DV_Number.php]: Note that the regex below can explode if one of these delimiters is longer than 1 character.

File: includes/datavalues/SMW_DV_Number.php [hunk 3]
```diff
@@ -71,10 +71,20 @@ static protected function parseNumberValue( $value, &$number, &$unit ) {
...
+		$regex = '/([-+]?\s*(?:' .
+				// Either numbers like 10,000.99 that start with a digit
+				'\d+(?:\\' . $kiloseparator . '\d\d\d)*(?:\\' . $decseparator . '\d+)?' .
+				// or numbers like .001 that start with the decimal separator
+				'|\\' . $decseparator . '\d+' .
+				')\s*(?:[eE][-+]?\d+)?)/u';
+
+		$parts = preg_split(
+			$regex,
+			trim( str_replace( array( '&nbsp;', '&#160;', '&thinsp;', ' ' ), '', $value ) ),
```
Comment (thiemowmde) [inline includes/datavalues/SMW_DV_Number.php]: This looks like you should use `html_entity_decode`.

**Pull request merged**